### ISSUE TITLE
[Linux] spontanious UI corruption  

### ISSUE THREAD

**Issue by literallyunplayable** [ValveSoftware/Dota-2#380]
In reborn, opening the reborn overlay while in game, as well as several other in game events causes the following effect.
![2015-09-09_00004](https://cloud.githubusercontent.com/assets/8964525/9770130/2c748a90-56e3-11e5-979e-5ac338d7f823.jpg)
![2015-09-09_00006](https://cloud.githubusercontent.com/assets/8964525/9770131/2e77571e-56e3-11e5-90aa-3ce64e2b08e4.jpg)

As you can see by the images, text and images become garbled, occasionally when this happens tab labels such as "heroes" become replaced with random, misplaced strings from elsewhere in the client.

The effect is similar to issue #120 but the workarounds posted in that thread was not effective in resolving it. I believe it may be a different issue.

I tried to use dump_panorama_events but then could not copy-paste the output even with the reborn client still open. Are log files stored somewhere? Please offer instruction on how to log console output.

Processor Information:
    Vendor:  AuthenticAMD
    CPU Family:  0x15
    CPU Model:  0x1
    CPU Stepping:  0x2
    CPU Type:  0x0
    Speed:  3100 Mhz
    8 logical processors
    8 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Supported
    SSE41:  Supported
    SSE42:  Supported

Network Information:
    Network Speed:  

Operating System Version:
    Ubuntu 15.04 (64 bit)
    Kernel Name:  Linux
    Kernel Version:  3.19.0-28-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11701000
    X Window Manager:  Compiz
    Steam Runtime Version:  steam-runtime-release_2015-06-12

Video Card:
    Driver:  ATI Technologies Inc. AMD Radeon HD 6700 Series 

```
Driver Version:  4.4.13374 Compatibility Profile Context 15.20.1013
OpenGL Version: 4.4
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 60 Hz
VendorID:  0x1002
DeviceID:  0x68ba
Number of Monitors:  1
Number of Logical Video Cards:  1
Primary Display Resolution:  1920 x 1080
Desktop Resolution: 1920 x 1080
Primary Display Size: 18.78" x 10.55"  (21.54" diag)
                                        47.7cm x 26.8cm  (54.7cm diag)
Primary VRAM Not Detected
```

Sound card:
    Audio device: Realtek ALC889

Memory:
    RAM:  3933 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Total Hard Disk Space Available:  116044 Mb
    Largest Free Hard Disk Block:  34822 Mb

Installed software:

Recent Failure Reports:
    Wed Sep  9 17:47:36 2015 GMT: file ''/tmp/dumps/crash_20150909104037_1.dmp'', upload yes: ''Discarded=1''
    Wed Sep  9 17:52:23 2015 GMT: file ''/tmp/dumps/crash_20150909104918_1.dmp'', upload yes: ''Discarded=1''
    Wed Sep  9 17:55:47 2015 GMT: file ''/tmp/dumps/crash_20150909105254_1.dmp'', upload yes: ''Discarded=1''

**Comment by conmute** [comment]
I do have the same issue. I use Ubuntu Trusty Tahr (14.04)

My notebook is Dell Inspiron 15R.

video graphics:

```
$ lspci -nnk | grep VGA -A1
00:02.0 VGA compatible controller [0300]: Intel Corporation 3rd Gen Core processor Graphics Controller [8086:0166] (rev 09)
    Subsystem: Dell Device [1028:056a]
--
01:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Thames [Radeon HD 7500M/7600M Series] [1002:6840]
    Subsystem: Dell Radeon HD 7670M [1028:056a]
$ lspci -nnk | grep VGA -A1
00:02.0 VGA compatible controller [0300]: Intel Corporation 3rd Gen Core processor Graphics Controller [8086:0166] (rev 09)
    Subsystem: Dell Device [1028:056a]
--
01:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Thames [Radeon HD 7500M/7600M Series] [1002:6840]
    Subsystem: Dell Radeon HD 7670M [1028:056a]
```

CPU

```
$ cat /proc/cpuinfo
processor   : 0
vendor_id   : GenuineIntel
cpu family  : 6
model       : 58
model name  : Intel(R) Core(TM) i7-3612QM CPU @ 2.10GHz
stepping    : 9
microcode   : 0x12
cpu MHz     : 2101.000
cache size  : 6144 KB
physical id : 0
siblings    : 8
core id     : 0
cpu cores   : 4
apicid      : 0
initial apicid  : 0
fpu     : yes
fpu_exception   : yes
cpuid level : 13
wp      : yes
flags       : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm ida arat epb xsaveopt pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms
bogomips    : 4190.39
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:

processor   : 1
vendor_id   : GenuineIntel
cpu family  : 6
model       : 58
model name  : Intel(R) Core(TM) i7-3612QM CPU @ 2.10GHz
stepping    : 9
microcode   : 0x12
cpu MHz     : 2101.000
cache size  : 6144 KB
physical id : 0
siblings    : 8
core id     : 0
cpu cores   : 4
apicid      : 1
initial apicid  : 1
fpu     : yes
fpu_exception   : yes
cpuid level : 13
wp      : yes
flags       : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm ida arat epb xsaveopt pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms
bogomips    : 4190.39
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:

processor   : 2
vendor_id   : GenuineIntel
cpu family  : 6
model       : 58
model name  : Intel(R) Core(TM) i7-3612QM CPU @ 2.10GHz
stepping    : 9
microcode   : 0x12
cpu MHz     : 2101.000
cache size  : 6144 KB
physical id : 0
siblings    : 8
core id     : 1
cpu cores   : 4
apicid      : 2
initial apicid  : 2
fpu     : yes
fpu_exception   : yes
cpuid level : 13
wp      : yes
flags       : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm ida arat epb xsaveopt pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms
bogomips    : 4190.39
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:

processor   : 3
vendor_id   : GenuineIntel
cpu family  : 6
model       : 58
model name  : Intel(R) Core(TM) i7-3612QM CPU @ 2.10GHz
stepping    : 9
microcode   : 0x12
cpu MHz     : 2101.000
cache size  : 6144 KB
physical id : 0
siblings    : 8
core id     : 1
cpu cores   : 4
apicid      : 3
initial apicid  : 3
fpu     : yes
fpu_exception   : yes
cpuid level : 13
wp      : yes
flags       : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm ida arat epb xsaveopt pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms
bogomips    : 4190.39
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:

processor   : 4
vendor_id   : GenuineIntel
cpu family  : 6
model       : 58
model name  : Intel(R) Core(TM) i7-3612QM CPU @ 2.10GHz
stepping    : 9
microcode   : 0x12
cpu MHz     : 2101.000
cache size  : 6144 KB
physical id : 0
siblings    : 8
core id     : 2
cpu cores   : 4
apicid      : 4
initial apicid  : 4
fpu     : yes
fpu_exception   : yes
cpuid level : 13
wp      : yes
flags       : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm ida arat epb xsaveopt pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms
bogomips    : 4190.39
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:

processor   : 5
vendor_id   : GenuineIntel
cpu family  : 6
model       : 58
model name  : Intel(R) Core(TM) i7-3612QM CPU @ 2.10GHz
stepping    : 9
microcode   : 0x12
cpu MHz     : 2101.000
cache size  : 6144 KB
physical id : 0
siblings    : 8
core id     : 2
cpu cores   : 4
apicid      : 5
initial apicid  : 5
fpu     : yes
fpu_exception   : yes
cpuid level : 13
wp      : yes
flags       : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm ida arat epb xsaveopt pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms
bogomips    : 4190.39
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:

processor   : 6
vendor_id   : GenuineIntel
cpu family  : 6
model       : 58
model name  : Intel(R) Core(TM) i7-3612QM CPU @ 2.10GHz
stepping    : 9
microcode   : 0x12
cpu MHz     : 2101.000
cache size  : 6144 KB
physical id : 0
siblings    : 8
core id     : 3
cpu cores   : 4
apicid      : 6
initial apicid  : 6
fpu     : yes
fpu_exception   : yes
cpuid level : 13
wp      : yes
flags       : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm ida arat epb xsaveopt pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms
bogomips    : 4190.39
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:

processor   : 7
vendor_id   : GenuineIntel
cpu family  : 6
model       : 58
model name  : Intel(R) Core(TM) i7-3612QM CPU @ 2.10GHz
stepping    : 9
microcode   : 0x12
cpu MHz     : 2101.000
cache size  : 6144 KB
physical id : 0
siblings    : 8
core id     : 3
cpu cores   : 4
apicid      : 7
initial apicid  : 7
fpu     : yes
fpu_exception   : yes
cpuid level : 13
wp      : yes
flags       : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm ida arat epb xsaveopt pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms
bogomips    : 4190.39
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:

```

**Comment by ArnaudAndre** [comment]
conmute : you're using catalyst too right ?
Have you tried running dota on the integrated gpu ? 3rd Gen Core Graphics means HD Graphics 2500/4000 right ? That should allow you to run dota2 for a quick test, although framerate will be awful.

**Comment by jaredmcateer** [comment]
I am also having these problems on my desktop

```
Ubuntu 14.04 LTS
Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz
Asus P8Z77-V DELUXE
Advanced Micro Devices, Inc. [AMD/ATI] Pitcairn PRO [Radeon HD 7850]

18:17 $ LIBGL_DEBUG=verbose fglrxinfo 2>&1 | grep fglrx
libGL: AtiGetClientDriverName: 15.20.2 fglrx (screen 0)
```

**Comment by conmute** [comment]
@ArnaudAndre yes, i using catalyst, fglrx drivers.

When i switched to integrated graphics i started to have this isse #234 . when going back, all is ok.

More descruption about user expirience with this bug.

I will start with old dota 2, before few updates it was removed from users usage, i had such bug, i think it can be related somehow to this issue. After few matches,i had black boxes. Conection status, dialog messages, in fight the bar under hero and few more. Those black boxes where flickering and very unoying, after game restart all back to normal. 1 match after black boxes again.

In Dota 2 reborn i have this stuff when i navigate often in menu view. And revealing much faster when in game. With game restart all is back to normal. Graphic distortion differs, from simple header button distortion, to in game flkering. I will push new images when i return to game.

So for now, good play is only when i just do: go to game and simple wait for a battle. nothing more, no hard navigating in menu mode.

**Comment by ArnaudAndre** [comment]
Ah yes, for ubuntu 14.04 you'd need to get a PPA carrying a more recent mesa. Although with the 3rd report on this coming from a catalyst user there's even less doubt about the problem's source...

**Comment by jaredmcateer** [comment]
I've upgraded to 14.10 and the issue persists, I'll be upgrading to 15 today and will report back whether it makes a difference (I doubt it will as it seems clear it's something to do with the AMD proprietary drivers at this point.)

**Comment by danginsburg** [comment]
Yesterday AMD released Catalyst 15.9, we heard similar issues such as #424 were fixed by it.  Please give it a try and report back: http://support.amd.com/en-us/download/desktop?os=Linux+x86

**Comment by simplystic** [comment]
Had same problems as mentioned here, installed new drivers on my Ubuntu 15.04 at it fixed EVERYTHING for me.

**Comment by danginsburg** [comment]
OK, so it sounds like this was a bug with Catalyst 15.7 that has been fixed by Catalyst 15.9.  Closing.

**Comment by jaredmcateer** [comment]
Confirming, upgraded to Catalyst 15.9 and my issues have been resolved.
